engines/misc/src/main/java/org/alfresco/transform/misc/transformers/HtmlParserContentTransformer.java

```diff
@@ -1,203 +1,229 @@
+ * GNU Lesser General Public License for more details.
+ * -
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.transform.misc.transformers;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.net.URLConnection;
+import java.nio.charset.Charset;
+import java.nio.charset.IllegalCharsetNameException;
+import java.util.Map;
+
+import org.alfresco.transform.base.TransformManager;
+import org.alfresco.transform.base.util.CustomTransformerFileAdaptor;
+import static org.alfresco.transform.common.RequestParamMap.SOURCE_ENCODING;
+import org.alfresco.transform.config.TransformConfig;
+import org.alfresco.transform.config.TransformOption;
+import org.alfresco.transform.config.TransformOptionValue;
+import org.alfresco.transform.config.Transformer;
+import org.alfresco.transform.misc.MiscTransformEngine;
+import org.htmlparser.Parser;
+import org.htmlparser.beans.StringBean;
+import org.htmlparser.util.ParserException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import jakarta.annotation.PostConstruct;
+
+/**
+ * Content transformer which wraps the HTML Parser library for parsing HTML content.
+ *
+ * <p>
+ * This code is based on a class of the same name originally implemented in alfresco-repository.
+ * </p>
+ *
+ * <p>
+ * Since HTML Parser was updated from v1.6 to v2.1, META tags defining an encoding for the content via http-equiv=Content-Type will ONLY be respected if the encoding of the content item itself is set to ISO-8859-1.
+ * </p>
+ *
+ * <p>
+ * Tika Note - could be converted to use the Tika HTML parser, but we'd potentially need a custom text handler to replicate the current settings around links and non-breaking spaces.
+ * </p>
+ *
+ * @author Derek Hulley
+ * @author eknizat
+ * @see <a href="http://htmlparser.sourceforge.net/">http://htmlparser.sourceforge.net</a>
+ * @see org.htmlparser.beans.StringBean
+ * @see <a href="http://sourceforge.net/tracker/?func=detail&aid=1644504&group_id=24399&atid=381401">HTML Parser</a>
+ */
+@Component
+public class HtmlParserContentTransformer implements CustomTransformerFileAdaptor
+{
+    private static final Logger logger = LoggerFactory.getLogger(
+            HtmlParserContentTransformer.class);
+
+    private static final String HTML_COLLAPSE = "collapseHtml";
+    private static final String HTML_OPTIONS = "htmlOptions";
+    private boolean COLLAPSE_HTML = true;
+
+    @Autowired
+    private MiscTransformEngine transformEngine;
+
+    @Override
+    public String getTransformerName()
+    {
+        return "html";
+    }
+
+    @PostConstruct
+    public void init() {
```

Comment:
This initialization can be removed since the option value will be obtained at request level, not at bean initialization.